### PR TITLE
AlertTab: some ui updates

### LINKED ISSUE #23912: AlertTab: Remove usage of EditorTabBody move actions down
* Replace EditorTabBody with a CustomScrollbar
* Move actions down to the bottom of the view and make them normal buttons (delete into danger).
* Move test rule view and alert history view to a modal

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
This closes the following issue: #23912

**Todo:**
- [x] When running "state history" there is an "clear history" button that will open another modal. This needs to be fixed since modal on modal is not a good pattern.
- [x] Verify modal icons, title is correct. @torkelo

### CODE REVIEW

**Comment by torkelo** [comment]
Mabye the clear history can use our ConfirmButton?  https://developers.grafana.com/ui/canary/index.html?path=/story/buttons-confirmbutton--default

**Comment by mckn** [comment]
> Mabye the clear history can use our ConfirmButton? https://developers.grafana.com/ui/canary/index.html?path=/story/buttons-confirmbutton--default

yeah, that would work!

**Comment by mckn** [comment]
@torkelo I think it is good to go. Just need your verification that it is looking as you planned.

File: public/app/features/alerting/AlertTab.tsx [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
-import { Alert, Button, IconName } from '@grafana/ui';
+import { Alert, Button, IconName, CustomScrollbar, Container, HorizontalGroup, ConfirmModal, Modal } from '@grafana/ui';
 import { selectors } from '@grafana/e2e-selectors';
 import { AngularComponent, getAngularLoader, getDataSourceSrv } from '@grafana/runtime';
-
-import appEvents from 'app/core/app_events';
 import { getAlertingValidationMessage } from './getAlertingValidationMessage';
 
-import { EditorTabBody, EditorToolbarView } from '../dashboard/panel_editor/EditorTabBody';
+// import { EditorToolbarView } from '../dashboard/panel_editor/EditorTabBody';
```
[the remark this answers is not in the document]
Comment (mckn) [reply public/app/features/alerting/AlertTab.tsx:9]: ah, good catch!

File: public/app/features/alerting/AlertTab.tsx [hunk 2]
```diff
@@ -205,19 +228,36 @@ class UnConnectedAlertTab extends PureComponent<Props, State> {
...
+                  severity={AppNotificationSeverity.Error}
+                  title="Transformations are not supported in alert queries"
+                />
+              )}
+
+              <div ref={element => (this.element = element)} />
+              {alert && (
+                <HorizontalGroup>
+                  <Button onClick={() => this.onToggleModal('showStateHistory')}>State history</Button>
+                  <Button onClick={() => this.onToggleModal('showTestRule')}>Test rule</Button>
```
Comment (dprokop) [inline public/app/features/alerting/AlertTab.tsx:246]: Think we can use secondary buttons here to be in line with other tabs:
```suggestion
                  <Button onClick={() => this.onToggleModal('showStateHistory')} variant="secondary">State history</Button>
                  <Button onClick={() => this.onToggleModal('showTestRule')} variant="secondary">Test rule</Button>
```

**Comment by dprokop** [comment]
image

Got this when clearing the history

**Review by dprokop — APPROVED** [review]
Iz good :D

**Comment by dprokop** [comment]
@mckn pushed prettier fix

**Review by torkelo — APPROVED** [review]
Looks good!

File: public/app/features/alerting/AlertTab.tsx [hunk 3]
```diff
@@ -186,6 +143,74 @@ class UnConnectedAlertTab extends PureComponent<Props, State> {
...
+  renderTestRule = () => {
+    if (!this.state.showTestRule) {
+      return null;
+    }
+
+    const { panel, dashboard } = this.props;
+    const onDismiss = () => this.onToggleModal('showTestRule');
+
+    return (
+      <Modal isOpen={true} icon="play" title="Testing rule" onDismiss={onDismiss} onClickBackdrop={onDismiss}>
```
Comment (torkelo) [inline public/app/features/alerting/AlertTab.tsx:155]: icon: bug

Comment (dprokop) [inline public/app/features/alerting/AlertTab.tsx:155]:
```suggestion
      <Modal isOpen={true} icon="bug" title="Testing rule" onDismiss={onDismiss} onClickBackdrop={onDismiss}>
```

**Pull request merged**